src/ninja.cc

```diff
@@ -909,6 +912,38 @@ void printCompdb(const char* const directory, const Edge* const edge,
   printf("\"\n  }");
 }
 
+bool GetAllDependentEdges(Node* node, std::vector<Edge*>* depend_edges,
+                          std::set<Node*>* visited_nodes=NULL,
+                          std::set<Edge*>* visited_edges=NULL) {
+  if (visited_nodes == NULL || visited_edges == NULL) {
+    std::set<Node*> visited_nodes_;
+    std::set<Edge*> visited_edges_;
+    return GetAllDependentEdges(node, depend_edges, &visited_nodes_, &visited_edges_);
+  }
+
+  assert(node);
+  assert(depend_edges);
+
+  if (visited_nodes->count(node)) {
+    return true;
+  }
+  visited_nodes->insert(node);
```

Comment:
nit: use the result of insert() to avoid the count() call above and perform one lookup instead of two, e.g.:

```
if (!visited_nodes->insert(node).second)
  return true;
```